OpenHPL/Waterway/SurgeTanks.mo

```diff
@@ -60,21 +66,57 @@ equation
   der(m) = m_dot "Mass balance";
   der(M) = M_dot+F "Momentum balance";
 
-  m = para.rho * A * l+m_a;
+  if surge_tank_type == OpenHPL.Types.SurgeTank.SimpleST then
+    m = para.rho * A * l;
+    p_t = para.p_a;
+    F_f = Functions.DarcyFriction.Friction(v, D, l, para.rho, para.mu, eps);
+  elseif  surge_tank_type == OpenHPL.Types.SurgeTank.AirCushionST then
+    m = para.rho * A * l+m_a;
+    p_t = p_t0*((L-h_0/cos_theta)/(L-l))^para.gamma_air;
+    F_f = Functions.DarcyFriction.Friction(v, D, l, para.rho, para.mu, eps);
+  /*elseif  surge_tank_type == OpenHPL.Types.SurgeTank.ThrottleValveST then
+    m = para.rho * A * l+m_a;
+    p_t = p_t0*((L-h_0/cos_theta)/(L-l))^para.gamma_air;
+    */
+  elseif  surge_tank_type == OpenHPL.Types.SurgeTank.SharpOrificeST then
+    m = para.rho * A * l;
+    p_t = para.p_a;
+    F_f = Functions.DarcyFriction.Friction(v, D, l, para.rho, para.mu, eps)+ phi*0.5 * para.rho * abs(v) * v;
+    if v>=0 then
+      phi = Functions.Fitting.FittingPhi(v,D,D_so,L,theta,para.rho,para.mu,para.eps,SharpOrifice);
+    else
+      phi = Functions.Fitting.FittingPhi(v,D_so,D,L,theta,para.rho,para.mu,para.eps,SharpOrifice);
+    end if;
+
```

Comment:
I would probably write the if then elses in such a way that the special cases are handled by ifs and the simple case is then the else default.